fog/types/src/common.rs

```diff
@@ -39,6 +47,16 @@ impl BlockRange {
     pub fn overlaps(&self, other: &BlockRange) -> bool {
         self.start_block < other.end_block && other.start_block < self.end_block
     }
+
+    /// Returns the length of the BlockRange, i.e. the number of blocks.
+    pub fn len(&self) -> usize {
+        (self.end_block - self.start_block) as usize
```

Comment:
It seems safer to keep this as a `u64` -- `usize` could be 32 bits
```suggestion
    pub fn len(&self) -> u64 {
        self.end_block - self.start_block
```

Reply:
This got changed in response to a comment I made, https://github.com/mobilecoinfoundation/mobilecoin/pull/2845#discussion_r1017221789
> :question: Doing some quick spot checks it seems most len() functions return a usize? i'm guessing this internal start and end are u64 due to prost, but it seems plausible the length here could be usize.
>
> Not really sure on this. I was mainly looking at through std for len() so it could just be std

I'm fine if we think u64 is the more robust option. My argument was mainly for consistency with other APIs.
For usize to be an issue on 32 bit systems the block range would have to exceed 4billion blocks.  Which could be probable as the chain grows. Not sure the likely hood or if this code would actually ever be used on 32 bit systems.  My understanding is the enclave code is more or less limited to 64 bit sgx platforms, but this isn't guaranteed to be enclave only.

Reply:
A `BlockRange` greater than 4 billion blocks seems more plausible than an in-memory container with that many entries. So it's safer to use `u64`.

With that said, I don't feel strongly about this, and y'all have more context about this code, so I won't block approval on this.

Reply:
Yeah, I think `u64` is more robust because it ensures we never have to convert `u64` to `usize`. Done.